src/routes/add-custom-discipline/AddCustomDisciplineScreen.tsx

```diff
@@ -111,21 +89,20 @@ const AddCustomDiscipline = ({ navigation }: AddCustomDisciplineScreenProps): JS
         <Container>
           <CustomDisciplineHeading>{labels.addCustomDiscipline.heading}</CustomDisciplineHeading>
           <Description>{labels.addCustomDiscipline.description}</Description>
-          <InputContainer errorMessage={errorMessage}>
-            <StyledTextInput
+          <InputContainer>
+            <CustomTextInput
+              errorMessage={errorMessage}
               placeholder={labels.addCustomDiscipline.placeholder}
-              placeholderTextColor={theme.colors.placeholder}
               value={code}
               onChangeText={setCode}
+              clearable
+              rightContainer={
+                <TouchableOpacity onPress={() => setShowQRCodeOverlay(true)}>
+                  <QRCodeIcon accessibilityLabel='qr-code-scanner' width={wp('6%')} height={wp('6%')} />
```

Comment:
styling of the qr-code icon is now wrong. The padding on the right is too small.